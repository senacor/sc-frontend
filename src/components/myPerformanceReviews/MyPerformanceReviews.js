import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import { Link } from 'react-router-dom';

import MyPerformanceReviewsList from './MyPerformanceReviewsList';
import RequestPerformanceReview from './RequestPerformanceReview';
import PerformanceReviewDetail from '../pr/PerformanceReviewDetail';

class MyPerformanceReviews extends Component {
  render() {
    return (
      <div>
        <Typography variant="display1" paragraph>
          Performance Reviews
        </Typography>
        <Grid container spacing={8}>
          <Grid item xs={12} sm={12} md={12} lg={3} xl={3}>
            <RequestPerformanceReview />
            <Link to={'/availabilityview'} style={{ textDecoration: 'none' }}>
              <Button color="primary">
                <Icon>add</Icon>
                Termin finden
              </Button>
            </Link>
            <Hidden mdDown>
              <MyPerformanceReviewsList highlightPrDetail={true} />
            </Hidden>
            <Hidden smUp>
              <MyPerformanceReviewsList highlightPrDetail={false} />
            </Hidden>
          </Grid>
          <Hidden mdDown>
            <Grid item lg={9} xl={9}>
              <PerformanceReviewDetail />
            </Grid>
          </Hidden>
        </Grid>
      </div>
    );
  }
}

export default MyPerformanceReviews;
