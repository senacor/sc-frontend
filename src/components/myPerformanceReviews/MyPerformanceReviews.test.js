import React from 'react';
import { shallow } from 'enzyme';
import MyPerformanceReviews from './MyPerformanceReviews';

describe('MyPerformanceReviews', () => {
  it('should match snapshot', () => {
    let cut = shallow(<MyPerformanceReviews />);

    expect(cut).toMatchSnapshot();
  });
});
