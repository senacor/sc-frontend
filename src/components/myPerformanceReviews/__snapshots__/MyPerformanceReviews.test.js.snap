// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPerformanceReviews should match snapshot 1`] = `
<div>
  <WithStyles(Typography)
    paragraph={true}
    variant="display1"
  >
    Performance Reviews
  </WithStyles(Typography)>
  <WithStyles(Grid)
    container={true}
    spacing={8}
  >
    <WithStyles(Grid)
      item={true}
      lg={3}
      md={12}
      sm={12}
      xl={3}
      xs={12}
    >
      <Connect(RequestPerformanceReview) />
      <Connect(WithStyles(MyPerformanceReviewsList))
        highlightPrDetail={true}
      />
    </WithStyles(Grid)>
    <Hidden
      implementation="js"
      lgDown={false}
      lgUp={false}
      mdDown={true}
      mdUp={false}
      smDown={false}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Grid)
        item={true}
        lg={9}
        xl={9}
      >
        <Connect(PerformanceReviewDetail) />
      </WithStyles(Grid)>
    </Hidden>
  </WithStyles(Grid)>
</div>
`;
